src/evaluate/evaluation_suite/__init__.py

```diff
@@ -0,0 +1,131 @@
+        if type(self.task_type) is not str:
+            raise ValueError(f"'task_type' must be type 'str', got {type(self.task_type)}")
+        if type(self.data) not in [Dataset, str]:
+            raise ValueError(
+                f"'data' must be an already-instantiated Dataset object or type 'str', got {type(self.data)}"
+            )
+        if self.subset and type(self.subset) is not str:
+            raise ValueError(f"'subset' must be type 'str', got {type(self.subset)}")
+        if self.split and type(self.split) is not str:
+            raise ValueError(f"'split' must be type 'str', got {type(self.split)}")
+        if self.data_preprocessor and not callable(self.data_preprocessor):
+            raise ValueError(f"'data_preprocessor' must be a Callable', got {self.data_preprocessor}")
+        if self.args_for_task and type(self.args_for_task) is not dict:
+            raise ValueError(f"'args_for_task' must be type 'dict', got {type(self.args_for_task)}")
+
+
+def import_main_class(module_path):
+    """Import a module at module_path and return the EvaluationSuite class"""
+    module = importlib.import_module(module_path)
+
+    module_main_cls = None
+    for name, obj in module.__dict__.items():
+        if isinstance(obj, type) and obj.__name__ == "Suite":
+            if inspect.isabstract(obj):
+                continue
+            module_main_cls = obj
+            break
+
+    return module_main_cls
+
+
+class EvaluationSuite:
+    """
+    This class instantiates an evaluation suite made up of multiple tasks, where each task consists of a dataset and
+    an associated metric, and runs evaluation on a model or pipeline. Evaluation suites can be a Python script found
+    either locally or uploaded as a Space on the Hugging Face Hub.
+    Usage:
+    ```python
+    from evaluate import EvaluationSuite
+    suite = EvaluationSuite.load('mathemakitten/glue-evaluation-suite')
+    results = suite.run("gpt2")
+    ```
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.hasher = Hasher()
+
+    @staticmethod
+    def load(
+        path: str,
+        download_mode: Optional[DownloadMode] = None,
+        revision: Optional[Union[str, Version]] = None,
+        download_config: Optional[DownloadConfig] = None,
+    ):
+        download_mode = DownloadMode(download_mode or DownloadMode.REUSE_DATASET_IF_EXISTS)
+        evaluation_module = evaluation_module_factory(
+            path, module_type=None, revision=revision, download_config=download_config, download_mode=download_mode
+        )
+        name = Path(path).stem
+        evaluation_cls = import_main_class(evaluation_module.module_path)
+        evaluation_instance = evaluation_cls(name)
+
+        return evaluation_instance
+
+    def __repr__(self):
+        self.tasks = [str(task) for task in self.suite]
+        return f'EvaluationSuite name: "{self.name}", ' f"Tasks: {self.tasks})"
+
+    def assert_suite_nonempty(self):
```

Comment:
This would be a great case to test: check if the error is raised when an empty suite is passed.